nodl_python/nodl/exception.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Optional
+
+from lxml import etree
+
+
+class NoDLException(Exception):
+    """Base class for all NoDL exceptions."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+class InvalidNoDLException(NoDLException):
```

Comment:
```suggestion
class InvalidNoDLError(NoDLException):
```

Following Python's established pattern here of ending potentially user-facing exceptions with Error.

Reply:
Updated